src/Umbraco.Core/Configuration/ClientDependencyConfiguration.cs

```diff
@@ -49,5 +53,57 @@ internal bool IncreaseVersionNumber()
 
             return false;
         }
+
+        /// <summary>
+        /// Clears the temporary files stored for the ClientDependency folder
+        /// </summary>
+        /// <param name="currentHttpContext"></param>
+        internal bool ClearTempFiles(HttpContextBase currentHttpContext)
+        {
+            var cdfTempDirectories = new HashSet<string>();
+            foreach (BaseCompositeFileProcessingProvider provider in ClientDependencySettings.Instance
+                .CompositeFileProcessingProviderCollection)
+            {
+                var path = provider.CompositeFilePath.FullName;
+                cdfTempDirectories.Add(path);
+            }
+
+            try
+            {
+                var relativePath = currentHttpContext.Server.MapPath(XmlFileMapper.FileMapVirtualFolder);
+                if (relativePath != null)
+                {
+                    var directoryInfo = new DirectoryInfo(relativePath);

```

Comment:
We don't actually need to create a DirectoryInfo object here at all, the `relativePath` variable is already the full path